tests/test_basic.py

```diff
@@ -216,12 +216,25 @@ def test_add_node(rook_cluster):
     i = 0
     while osds != workers_new:
         if i == 90:
-            pytest.fail("rook did not add an additional osd-node")
-            break
+            rook_cluster.kubernetes.hardware.node_remove(node)
+            pytest.fail("rook did not add an additional osd-node."
+                        f"Removed node {node} again")
         time.sleep(10)
         osds = rook_cluster.get_number_of_osds()
         i += 1
 
+    # now remove the node again
+    workers_current = len(rook_cluster.kubernetes.hardware.workers)
+    workers_expected = workers_current - 1
+    i = 0
+    rook_cluster.kubernetes.hardware.node_remove(node)
+    while workers_expected != workers_current:
+        if i == 10:
+            pytest.fail(f"Was not able to remove node {node}")
+        time.sleep(10)
+        workers_current = len(rook_cluster.kubernetes.hardware.workers)
+        i += 1
```

Comment:
Should probably wait for the number of OSD's to be restored to where is expected

Reply:
Done